### ISSUE TITLE
Unit test for ResilientTransaction example in resilient EF Core Connection.

### ISSUE THREAD

**Issue by sajanmittal** [dotnet/docs#17655]
for ResilientTransaction class example.

public class ResilientTransaction
{
    private DbContext _context;
    private ResilientTransaction(DbContext context) =>
        _context = context ?? throw new ArgumentNullException(nameof(context));

    public static ResilientTransaction New (DbContext context) =>
        new ResilientTransaction(context);

    public async Task ExecuteAsync(Func<Task> action)
    {
        // Use of an EF Core resiliency strategy when using multiple DbContexts
        // within an explicit BeginTransaction():
        // https://docs.microsoft.com/ef/core/miscellaneous/connection-resiliency
        var strategy = _context.Database.CreateExecutionStrategy();
        await strategy.ExecuteAsync(async () =>
        {
            using (var transaction = _context.Database.BeginTransaction())
            {
                await action();
                transaction.Commit();
            }
        });
    }
}

How to cover Unit Test for ExecuteAsync. DO we need to mock context Database and Execution Strategy?

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 57452ceb-e885-d193-d813-d7baf7291cbc
* Version Independent ID: 6ad8c3bc-fdef-1e24-175d-ca7c7b9a98ce
* Content: [Implement resilient Entity Framework Core SQL connections](https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/implement-resilient-entity-framework-core-sql-connections)
* Content Source: [docs/architecture/microservices/implement-resilient-applications/implement-resilient-entity-framework-core-sql-connections.md](https://github.com/dotnet/docs/blob/master/docs/architecture/microservices/implement-resilient-applications/implement-resilient-entity-framework-core-sql-connections.md)
* Product: **dotnet-architecture**
* Technology: **microservices**
* GitHub Login: @nishanil
* Microsoft Alias: **nanil**

**Comment by sughosneo** [comment]
Hi @sajanmittal , thanks for reaching out. 

Method `ExecuteAsync(Func<Task> action)` of class `ResilientTransaction` is calling `ExecuteAsync()` method of 'IExecutionStrategy', 
you would atleast need to create a mock object of type `IExecutionStrategy` so that you can call `Setup` method on that. 

Hope this helps !

**Comment by sughosneo** [comment]
Hi @sajanmittal , I am closing this issue as of now. If you have any further questions, please feel free to reopen it.

Thank you.

**Comment by Xerxes0198** [comment]
Hey, did anyone have any success with @sughosneo's suggestion?

I tried mocking the IExecutionStrategy class, and setting up the object as best I knew how, and I got only so far.
`    Castle.DynamicProxy.InvalidProxyConstructorArgumentsException : Can not instantiate proxy of class: Microsoft.EntityFrameworkCore.Infrastructure.DatabaseFacade.
    Could not find a parameterless constructor.`

My setup looks a little like this:
`dbContext.Setup(x => x.Database.CreateExecutionStrategy()).Returns(executionStrategy.Object);`

Any advice on this would be greatly appreciated.
